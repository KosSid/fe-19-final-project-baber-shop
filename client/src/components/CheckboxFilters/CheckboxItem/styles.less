.checkbox-container {
  label, input {
    cursor: pointer;
  }
  .checkbox-label {
    padding-left: 5px;
    font-size: 18px;
  }
  input[type='checkbox']{
    padding: 5px;
    background-color: #FFFFFF;
    border: 1px solid #000000;
    border-radius: 2px;
    box-shadow: none;
    -webkit-appearance: none;
    appearance: none;
  }
  input[type='checkbox']:checked{
    background-color: #000000;
  }
  input[type='checkbox']:focus{
    outline: none;
  }
  .checkbox-group {
    &__item{
      padding-top: 3px;
    }
  }
}

@media screen and (max-width: 585px){

  .checkbox-container .checkbox-label{
    font-size: 24px;
  }
  .checkbox-container input[type=checkbox]{
    padding: 10px;
  }
  .checkbox-group__item{
    margin: 5px 0;
  }
}

