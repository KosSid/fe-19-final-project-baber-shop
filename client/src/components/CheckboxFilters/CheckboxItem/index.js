import React, {useState, useEffect} from 'react';
import './styles.less';
import Ajax from "../../../services/Ajax";

const CheckboxItem = ({type, parsedUrl}) => {

    const [checkboxNames, setNames] = useState([]);

    useEffect(() => {
        async function fetch() {
            const result = await Ajax.get(`/filters/${type}`);
            setNames(result);
        }

        fetch()
    }, [type]);

    const names = checkboxNames.map(item => {
        return item.name
    })

    const filtersNamesUrl = Object.values(parsedUrl).flat(Infinity);

    return (
        <>{
            names.map(item =>
                <div className='checkbox-group__item' key={item}>
                    <input className='item-filter' data-type={type} defaultChecked={filtersNamesUrl.includes(item)} type="checkbox" id={item} name={item}/>
                    <label className='checkbox-label' htmlFor={item}>{item}</label>
                </div>
            )
        }
        </>
    )
}
export default CheckboxItem;