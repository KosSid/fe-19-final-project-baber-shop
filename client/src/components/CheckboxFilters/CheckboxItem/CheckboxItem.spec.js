import React from 'react';
import CheckboxItem from "./index";
import renderer from 'react-test-renderer';

test('CheckboxItem renders correctly', () => {
    const tree = renderer
        .create(<CheckboxItem id={"id"} checked={false} name={"name"} type={"type"} onChange={() => {}}/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

