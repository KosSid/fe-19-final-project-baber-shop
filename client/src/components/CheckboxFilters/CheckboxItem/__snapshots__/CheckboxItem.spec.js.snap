// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxItem is rendered true 1`] = `"<div class=\\"checkbox-group__item\\"><input class=\\"item-filter\\" type=\\"checkbox\\"><label class=\\"checkbox-label\\"></label></div>"`;

exports[`CheckboxItem renders correctly 1`] = `
<div
  className="checkbox-group__item"
>
  <input
    checked={false}
    className="item-filter"
    data-type="type"
    id="id"
    name="name"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    className="checkbox-label"
    htmlFor="id"
  >
    name
  </label>
</div>
`;
